src/mount_efs/__init__.py

```diff
@@ -578,6 +574,72 @@ def _validate_replacement_field_count(format_str, expected_ct):
     return dns_name
 
 
+def match_device(config, device):
+    """
+    Return the EFS id and the remote path to mount.
+
+    :param config: the current configuration
+    :param device: the device descriptor, separating an EFS id or a DNS name and the remote path to mount by a colon
+    :return: a two element tuple of the EFS id and the remote path to mount
+    """
+
+    # The device descriptor as specified separates the remote filesystem by the path to mount by a colon. Since colons
+    # are not allowed in either an EFS id or in a domain name, we can left-split once. (If left-splitting fails, the
+    # user didn't specify a path and we use '/' as the default.)
+    try:
+        remote, path = device.split(':', 1)
+    except ValueError:
+        remote = device
+        path = '/'
+
+    # The simplest case is that the remote is already the EFS id. If so, we return it as is.
+    if FS_ID_RE.match(remote):
+        return remote, path
+
+    # If the user did not specify an EFS id, we first check for the special case where the user supplied us with the
+    # FQDN of the EFS.
+    efs_fqdn_match = EFS_FQDN_RE.match(remote)
+    if efs_fqdn_match:
+        fs_id = efs_fqdn_match.group('fs_id')
+        expected_dns_name = get_dns_name(config, fs_id)
+        if remote == expected_dns_name:
+            return fs_id, path
+        else:
+            fatal_error(
+                'Fully qualified EFS domain name specified "%s", but it didn\'t match the expected value "%s"'
+                % (remote, expected_dns_name),
+                'EFS FQDN "%s" didn\'t match expected "%s"' % (remote, expected_dns_name)
+            )
+
+    # For the final case we now assume that the user specified a DNS resolvable name with a CNAME record that points to
+    # a valid EFS FQDN. To verify this, we'll use `socket.gethostbyname_ex()` which returns an alias-list for, among
+    # other things, a CNAME.
+    try:
+        primary, secondaries, _ = socket.gethostbyname_ex(remote)
+        hostnames = filter(lambda e: e is not None, [primary] + secondaries)
+    except socket.gaierror:
+        fatal_error(
+            'Failed to resolve "%s" - check that the specified DNS name is a CNAME record resolving to a valid EFS DNS '
+            'name' % remote,
+            'Failed to resolve "%s"' % remote
+        )
+
+    if not hostnames:
+        fatal_error(
+            'The specified domain name "%s" returned no entries, where at least one was expected' % remote
```

Comment:
The lay user would probably be confused what `entries` are, I'd do something like:
```
'The specified domain name "%s" did not resolve to an EFS mount target' % remote
```